components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go

```diff
@@ -93,38 +93,80 @@ func (spec *FunctionSpec) defaultReplicas(config *DefaultingConfig, fn *Function
 }
 
 func (spec *FunctionSpec) defaultFunctionResources(config *DefaultingConfig, fn *Function) {
-	resources := spec.ResourceConfiguration.Function.Resources
+	var resources *corev1.ResourceRequirements
+	if spec.ResourceConfiguration != nil && spec.ResourceConfiguration.Function != nil && spec.ResourceConfiguration.Function.Resources != nil {
+		resources = spec.ResourceConfiguration.Function.Resources
+	}
 	defaultingConfig := config.Function.Resources
 	resourcesPreset := mergeResourcesPreset(fn, FunctionResourcesPresetLabel, defaultingConfig.Presets, defaultingConfig.DefaultPreset, defaultingConfig.RuntimePresets)
 
-	spec.ResourceConfiguration.Function.Resources = defaultResources(&resources, resourcesPreset.RequestMemory, resourcesPreset.RequestCPU, resourcesPreset.LimitMemory, resourcesPreset.LimitCPU)
+	setFunctionResources(resources, resourcesPreset, spec)
+}
+
+func setFunctionResources(resources *corev1.ResourceRequirements, resourcesPreset ResourcesPreset, spec *FunctionSpec) {
```

Comment:
This function does 2 things:
- calculate some resources
- set those calculated resources

I propose following function, which only set calculated resources:
```
func setFunctionResources(spec *FunctionSpec, resources *corev1.ResourceRequirements){
...
```